pkg/controller/worker/controller.go

```diff
@@ -93,3 +101,34 @@ func add(mgr manager.Manager, options controller.Options, predicates []predicate
 		ToRequests: extensionshandler.SimpleMapper(ClusterToWorkerMapper(predicates), extensionshandler.UpdateWithNew),
 	})
 }
+
+func addStateUpdatingController(mgr manager.Manager, options controller.Options) error {
+	stateActuator := NewStateActuator(log.Log.WithName("worker-state-actuator"))
+	stateReconciler := NewStateReconciler(mgr, stateActuator)
+	addStateUpdatingControllerOptions := controller.Options{
+		MaxConcurrentReconciles: options.MaxConcurrentReconciles,
+		Reconciler:              stateReconciler,
+	}
+	predicates := []predicate.Predicate{
+		extensionspredicate.Or(
+			extensionspredicate.IsDeleting(),
```

Comment:
This triggers the watch when the worker is being deleted? I wonder why this predicate is needed? The state reconciler does not reconcile anyways [when there is a deletion timestamp](https://github.com/gardener/gardener-extensions/blob/4adcb350a6b15157bd27846528dfbe9d67b6c987/pkg/controller/worker/state_reconciler.go#L98).

Reply:
good point. Thanks